hydra/grammar/OverrideLexer.g4

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+// Regenerate lexer by running 'python setup.py antlr' at project root.
+// If you make changes here be sure to update the documentation (and update the grammar in command_line_syntax.md)
+lexer grammar OverrideLexer;
+
+
+// Re-usable Fragments.
+fragment DIGIT: [0-9];
+fragment INT_UNSIGNED: '0' | [1-9] (('_')? DIGIT)*;
+fragment CHAR: [a-zA-Z];
+fragment ID_: (CHAR|'_') (CHAR|DIGIT|'_')*;
+fragment INTERPOLATION_OPEN_: '${';
+fragment WS_: [ \t]+;
+fragment ESC_BACKSLASH_: '\\\\';  // escaped backslash
+
+/////////
+// KEY //
+/////////
+
+mode KEY;
+
+EQUAL: '=' WS_* -> mode(ARGS);
+
+TILDE: '~';
+PLUS: '+';
+AT: '@';
+COLON: ':';
+ATCOLON: '@:';
+SLASH: '/';
+
+ID: ID_;
+DOT_PATH: (ID | INT_UNSIGNED) ('.' (ID | INT_UNSIGNED))+;
+
+WS: WS_ -> skip;
+
+///////////////////
+// INTERPOLATION //
+///////////////////
+
+mode INTERPOLATION;
+
+INTERPOLATION_OPEN: INTERPOLATION_OPEN_ -> pushMode(INTERPOLATION);
+INTER_COLON: ':' WS_* -> type(COLON), mode(ARGS);
+INTERPOLATION_CLOSE: '}' -> popMode;
+
+DOT: '.';
+INTER_ID: ID_ -> type(ID);
+LIST_INDEX: INT_UNSIGNED;
+INTER_WS: WS_ -> skip;
+
+//////////
+// ARGS //
+//////////
+
+mode ARGS;
+
+ARGS_INTER_OPEN: INTERPOLATION_OPEN_ -> type(INTERPOLATION_OPEN), pushMode(INTERPOLATION);
+BRACE_OPEN: '{' WS_* -> pushMode(ARGS);  // must keep track of braces to detect end of interpolation
+BRACE_CLOSE: WS_* '}' -> popMode;
+
+// Numbers.
+
+fragment POINT_FLOAT: INT_UNSIGNED? '.' DIGIT (('_')? DIGIT)* | INT_UNSIGNED '.';
+fragment EXPONENT_FLOAT: (INT_UNSIGNED | POINT_FLOAT) [eE] [+-]? INT_UNSIGNED;
+FLOAT: [+-]? (POINT_FLOAT | EXPONENT_FLOAT | [Ii][Nn][Ff] | [Nn][Aa][Nn]);
+INT: [+-]? INT_UNSIGNED;
+
+// Other reserved keywords.
+
+BOOL:
+      [Tt][Rr][Uu][Ee]      // TRUE
+    | [Ff][Aa][Ll][Ss][Ee]; // FALSE
+
+NULL: [Nn][Uu][Ll][Ll];
+
+// Special characters.
+
+BRACKET_OPEN: '[' WS_*;
+BRACKET_CLOSE: WS_* ']';
+PARENTHESIS_OPEN: WS_* '(' WS_*;  // whitespaces before to allow `func (x)`
+PARENTHESIS_CLOSE: WS_* ')';
+
+COMMA: WS_* ',' WS_*;
+ARGS_COLON: WS_* ':' WS_* -> type(COLON);
+ARGS_EQUAL: WS_* '=' WS_* -> type(EQUAL);
+
+// Strings.
+
+ARGS_ID: ID_ -> type(ID);
+ESC: (ESC_BACKSLASH_ | '\\,' | '\\ ' | '\\\t')+;
+OTHER_CHAR: [/\-\\+.$*];  // other characters allowed in unquoted strings
+
+ARGS_WS: WS_ -> type(WS);
```

Comment:
So in order to address #873 I can't just skip whitespaces anymore. In order to avoid "polluting" the parser with lots of optional whitespaces, I made several lexer tokens "consume" them (like `COMMA`).